// @fluentui/react-datepicker-compat/lib-commonjs/components/DatePicker/defaults.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    defaultDatePickerErrorStrings: function() {
        return defaultDatePickerErrorStrings;
    },
    defaultDatePickerStrings: function() {
        return defaultDatePickerStrings;
    }
});
const _reactcalendarcompat = require("@fluentui/react-calendar-compat");
const defaultDatePickerStrings = {
    ..._reactcalendarcompat.defaultCalendarStrings,
    prevMonthAriaLabel: 'Go to previous month',
    nextMonthAriaLabel: 'Go to next month',
    prevYearAriaLabel: 'Go to previous year',
    nextYearAriaLabel: 'Go to next year',
    closeButtonAriaLabel: 'Close date picker'
};
const defaultDatePickerErrorStrings = {
    'invalid-input': 'Invalid date format',
    'out-of-bounds': 'Date is out of bounds',
    'required-input': 'Field is required'
};
