// @fluentui/react-datepicker-compat/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DatePicker: function() {
        return _DatePicker.DatePicker;
    },
    datePickerClassNames: function() {
        return _DatePicker.datePickerClassNames;
    },
    defaultDatePickerErrorStrings: function() {
        return _DatePicker.defaultDatePickerErrorStrings;
    },
    defaultDatePickerStrings: function() {
        return _DatePicker.defaultDatePickerStrings;
    },
    renderDatePicker_unstable: function() {
        return _DatePicker.renderDatePicker_unstable;
    },
    useDatePickerStyles_unstable: function() {
        return _DatePicker.useDatePickerStyles_unstable;
    },
    useDatePicker_unstable: function() {
        return _DatePicker.useDatePicker_unstable;
    }
});
const _DatePicker = require("./DatePicker");
