// @fluentui/react-calendar-compat/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AnimationDirection: function() {
        return _Calendar.AnimationDirection;
    },
    Calendar: function() {
        return _Calendar.Calendar;
    },
    CalendarDay: function() {
        return _CalendarDay.CalendarDay;
    },
    CalendarDayGrid: function() {
        return _CalendarDayGrid.CalendarDayGrid;
    },
    CalendarMonth: function() {
        return _CalendarMonth.CalendarMonth;
    },
    CalendarYear: function() {
        return _CalendarYear.CalendarYear;
    },
    DAYS_IN_WEEK: function() {
        return _utils.DAYS_IN_WEEK;
    },
    DEFAULT_CALENDAR_STRINGS: function() {
        return _utils.DEFAULT_CALENDAR_STRINGS;
    },
    DEFAULT_DATE_FORMATTING: function() {
        return _utils.DEFAULT_DATE_FORMATTING;
    },
    DEFAULT_DATE_GRID_STRINGS: function() {
        return _utils.DEFAULT_DATE_GRID_STRINGS;
    },
    DateRangeType: function() {
        return _utils.DateRangeType;
    },
    DayOfWeek: function() {
        return _utils.DayOfWeek;
    },
    FirstWeekOfYear: function() {
        return _utils.FirstWeekOfYear;
    },
    MonthOfYear: function() {
        return _utils.MonthOfYear;
    },
    TimeConstants: function() {
        return _utils.TimeConstants;
    },
    addDays: function() {
        return _utils.addDays;
    },
    addMonths: function() {
        return _utils.addMonths;
    },
    addWeeks: function() {
        return _utils.addWeeks;
    },
    addYears: function() {
        return _utils.addYears;
    },
    calendarClassNames: function() {
        return _Calendar.calendarClassNames;
    },
    calendarDayClassNames: function() {
        return _CalendarDay.calendarDayClassNames;
    },
    calendarDayGridClassNames: function() {
        return _CalendarDayGrid.calendarDayGridClassNames;
    },
    calendarPickerClassNames: function() {
        return _CalendarPicker.calendarPickerClassNames;
    },
    compareDatePart: function() {
        return _utils.compareDatePart;
    },
    compareDates: function() {
        return _utils.compareDates;
    },
    defaultCalendarStrings: function() {
        return _Calendar.defaultCalendarStrings;
    },
    extraCalendarDayGridClassNames: function() {
        return _CalendarDayGrid.extraCalendarDayGridClassNames;
    },
    getDatePartHashValue: function() {
        return _utils.getDatePartHashValue;
    },
    getDateRangeArray: function() {
        return _utils.getDateRangeArray;
    },
    getEndDateOfWeek: function() {
        return _utils.getEndDateOfWeek;
    },
    getMonthEnd: function() {
        return _utils.getMonthEnd;
    },
    getMonthStart: function() {
        return _utils.getMonthStart;
    },
    getStartDateOfWeek: function() {
        return _utils.getStartDateOfWeek;
    },
    getWeekNumber: function() {
        return _utils.getWeekNumber;
    },
    getWeekNumbersInMonth: function() {
        return _utils.getWeekNumbersInMonth;
    },
    getYearEnd: function() {
        return _utils.getYearEnd;
    },
    getYearStart: function() {
        return _utils.getYearStart;
    },
    isInDateRangeArray: function() {
        return _utils.isInDateRangeArray;
    },
    setMonth: function() {
        return _utils.setMonth;
    },
    useCalendarDayGridStyles_unstable: function() {
        return _CalendarDayGrid.useCalendarDayGridStyles_unstable;
    },
    useCalendarDayStyles_unstable: function() {
        return _CalendarDay.useCalendarDayStyles_unstable;
    },
    useCalendarMonthStyles_unstable: function() {
        return _CalendarMonth.useCalendarMonthStyles_unstable;
    },
    useCalendarPickerStyles_unstable: function() {
        return _CalendarPicker.useCalendarPickerStyles_unstable;
    },
    useCalendarStyles_unstable: function() {
        return _Calendar.useCalendarStyles_unstable;
    },
    useCalendarYearStyles_unstable: function() {
        return _CalendarYear.useCalendarYearStyles_unstable;
    }
});
const _Calendar = require("./Calendar");
const _CalendarDay = require("./CalendarDay");
const _CalendarDayGrid = require("./CalendarDayGrid");
const _CalendarMonth = require("./CalendarMonth");
const _CalendarPicker = require("./CalendarPicker");
const _CalendarYear = require("./CalendarYear");
const _utils = require("./utils");
