// @fluentui/react-calendar-compat/lib-commonjs/utils/animations.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DURATION_1: function() {
        return DURATION_1;
    },
    DURATION_2: function() {
        return DURATION_2;
    },
    DURATION_3: function() {
        return DURATION_3;
    },
    DURATION_4: function() {
        return DURATION_4;
    },
    EASING_FUNCTION_1: function() {
        return EASING_FUNCTION_1;
    },
    EASING_FUNCTION_2: function() {
        return EASING_FUNCTION_2;
    },
    FADE_IN: function() {
        return FADE_IN;
    },
    FADE_OUT: function() {
        return FADE_OUT;
    },
    SLIDE_DOWN_IN20: function() {
        return SLIDE_DOWN_IN20;
    },
    SLIDE_DOWN_OUT20: function() {
        return SLIDE_DOWN_OUT20;
    },
    SLIDE_LEFT_IN20: function() {
        return SLIDE_LEFT_IN20;
    },
    SLIDE_RIGHT_IN20: function() {
        return SLIDE_RIGHT_IN20;
    },
    SLIDE_UP_IN20: function() {
        return SLIDE_UP_IN20;
    },
    SLIDE_UP_OUT20: function() {
        return SLIDE_UP_OUT20;
    },
    TRANSITION_ROW_DISAPPEARANCE: function() {
        return TRANSITION_ROW_DISAPPEARANCE;
    }
});
const EASING_FUNCTION_1 = 'cubic-bezier(.1,.9,.2,1)';
const EASING_FUNCTION_2 = 'cubic-bezier(.1,.25,.75,.9)';
const DURATION_1 = '0.167s';
const DURATION_2 = '0.267s';
const DURATION_3 = '0.367s';
const DURATION_4 = '0.467s';
const FADE_IN = {
    from: {
        opacity: 0
    },
    to: {
        opacity: 1
    }
};
const FADE_OUT = {
    from: {
        opacity: 1
    },
    to: {
        opacity: 0,
        visibility: 'hidden'
    }
};
const SLIDE_DOWN_IN20 = {
    from: {
        pointerEvents: 'none',
        transform: 'translate3d(0, -20px, 0)'
    },
    to: {
        pointerEvents: 'auto',
        transform: 'translate3d(0, 0, 0)'
    }
};
const SLIDE_LEFT_IN20 = {
    from: {
        pointerEvents: 'none',
        transform: 'translate3d(20px, 0, 0)'
    },
    to: {
        pointerEvents: 'auto',
        transform: 'translate3d(0, 0, 0)'
    }
};
const SLIDE_RIGHT_IN20 = {
    from: {
        pointerEvents: 'none',
        transform: 'translate3d(-20px, 0, 0)'
    },
    to: {
        pointerEvents: 'auto',
        transform: 'translate3d(0, 0, 0)'
    }
};
const SLIDE_UP_IN20 = {
    from: {
        pointerEvents: 'none',
        transform: 'translate3d(0, 20px, 0)'
    },
    to: {
        pointerEvents: 'auto',
        transform: 'translate3d(0, 0, 0)'
    }
};
const SLIDE_DOWN_OUT20 = {
    from: {
        transform: 'translate3d(0, 0, 0)'
    },
    to: {
        transform: 'translate3d(0, 20px, 0)'
    }
};
const SLIDE_UP_OUT20 = {
    from: {
        transform: 'translate3d(0, 0, 0)'
    },
    to: {
        transform: 'translate3d(0, -20px, 0)'
    }
};
const TRANSITION_ROW_DISAPPEARANCE = {
    '100%': {
        height: '0px',
        overflow: 'hidden',
        width: '0px'
    },
    '99.9%': {
        height: '28px',
        overflow: 'visible',
        width: '100%'
    },
    '0%': {
        height: '28px',
        overflow: 'visible',
        width: '100%'
    }
};
