// @fluentui/react-calendar-compat/lib-commonjs/components/CalendarDayGrid/CalendarGridRow.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "CalendarGridRow", {
    enumerable: true,
    get: function() {
        return CalendarGridRow;
    }
});
const _interop_require_wildcard = require("@swc/helpers/_/_interop_require_wildcard");
const _react = /*#__PURE__*/ _interop_require_wildcard._(require("react"));
const _utils = require("../../utils");
const _CalendarGridDayCell = require("./CalendarGridDayCell");
const CalendarGridRow = (props)=>{
    const { ariaHidden, classNames, week, weeks, weekIndex, rowClassName, ariaRole, showWeekNumbers, firstDayOfWeek, firstWeekOfYear, navigatedDate, strings } = props;
    const weekNumbers = showWeekNumbers ? (0, _utils.getWeekNumbersInMonth)(weeks.length, firstDayOfWeek, firstWeekOfYear, navigatedDate) : null;
    const titleString = weekNumbers ? strings.weekNumberFormatString && strings.weekNumberFormatString.replace('{0}', `${weekNumbers[weekIndex]}`) : '';
    return /*#__PURE__*/ _react.createElement("tr", {
        role: ariaRole,
        "aria-hidden": ariaHidden,
        className: rowClassName,
        key: weekIndex + '_' + week[0].key
    }, showWeekNumbers && weekNumbers && /*#__PURE__*/ _react.createElement("th", {
        className: classNames.weekNumberCell,
        key: weekIndex,
        title: titleString,
        "aria-label": titleString,
        scope: "row"
    }, /*#__PURE__*/ _react.createElement("span", null, weekNumbers[weekIndex])), week.map((day, dayIndex)=>/*#__PURE__*/ _react.createElement(_CalendarGridDayCell.CalendarGridDayCell, {
            ...props,
            key: day.key,
            day: day,
            dayIndex: dayIndex
        })));
};
