// @fluentui/react-calendar-compat/lib-commonjs/utils/dateFormatting/dateFormatting.defaults.js
/**
 * Format date to a day string representation
 * @param date - input date to format
 */ "use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DEFAULT_CALENDAR_STRINGS: function() {
        return DEFAULT_CALENDAR_STRINGS;
    },
    DEFAULT_DATE_FORMATTING: function() {
        return DEFAULT_DATE_FORMATTING;
    },
    DEFAULT_DATE_GRID_STRINGS: function() {
        return DEFAULT_DATE_GRID_STRINGS;
    },
    formatDay: function() {
        return formatDay;
    },
    formatMonth: function() {
        return formatMonth;
    },
    formatMonthDayYear: function() {
        return formatMonthDayYear;
    },
    formatMonthYear: function() {
        return formatMonthYear;
    },
    formatYear: function() {
        return formatYear;
    }
});
const formatDay = (date)=>date.getDate().toString();
const formatMonthDayYear = (date, strings)=>strings.months[date.getMonth()] + ' ' + date.getDate() + ', ' + date.getFullYear();
const formatMonthYear = (date, strings)=>strings.months[date.getMonth()] + ' ' + date.getFullYear();
const formatMonth = (date, strings)=>strings.months[date.getMonth()];
const formatYear = (date)=>date.getFullYear().toString();
const DEFAULT_DATE_GRID_STRINGS = {
    months: [
        'January',
        'February',
        'March',
        'April',
        'May',
        'June',
        'July',
        'August',
        'September',
        'October',
        'November',
        'December'
    ],
    shortMonths: [
        'Jan',
        'Feb',
        'Mar',
        'Apr',
        'May',
        'Jun',
        'Jul',
        'Aug',
        'Sep',
        'Oct',
        'Nov',
        'Dec'
    ],
    days: [
        'Sunday',
        'Monday',
        'Tuesday',
        'Wednesday',
        'Thursday',
        'Friday',
        'Saturday'
    ],
    shortDays: [
        'S',
        'M',
        'T',
        'W',
        'T',
        'F',
        'S'
    ]
};
const DEFAULT_DATE_FORMATTING = {
    formatDay,
    formatMonth,
    formatYear,
    formatMonthDayYear,
    formatMonthYear
};
const DEFAULT_CALENDAR_STRINGS = {
    ...DEFAULT_DATE_GRID_STRINGS,
    goToToday: 'Go to today',
    weekNumberFormatString: 'Week number {0}',
    prevMonthAriaLabel: 'Previous month',
    nextMonthAriaLabel: 'Next month',
    prevYearAriaLabel: 'Previous year',
    nextYearAriaLabel: 'Next year',
    prevYearRangeAriaLabel: 'Previous year range',
    nextYearRangeAriaLabel: 'Next year range',
    closeButtonAriaLabel: 'Close',
    selectedDateFormatString: 'Selected date {0}',
    todayDateFormatString: "Today's date {0}",
    monthPickerHeaderAriaLabel: '{0}, change year',
    yearPickerHeaderAriaLabel: '{0}, change month',
    dayMarkedAriaLabel: 'marked'
};
